include/rmm/mr/device/pool_memory_resource.hpp

```diff
@@ -286,28 +330,17 @@ class pool_memory_resource final
    * @param maximum_size The optional maximum size for the pool
```

Comment:
**issue**: This docstring no longer matches the function. Since `initial_size` can now no longer be optional and it doesn't query `get_mem_info`. Perhaps:

```
@brief Allocate initial memory for the pool

@param initial_size The initial size for the pool
@param maximum_size The optional maximum size

@throws logic_error if maximum_size is provided and is smaller than initial_size
```
?